hydrus/parser/openapi_parser.py

```diff
@@ -114,103 +217,196 @@ def get_class_details(class_location: List[str],
+    dataType_ref_map["string"] = "https://schema.org/Text"
+    dataType_ref_map["long"] = "http://books.xmlschemata.org/relaxng/ch19-77199.html"
+    dataType_ref_map["float"] = "https://schema.org/Float"
+    dataType_ref_map["boolean"] = "https://schema.org/Boolean"
+    dataType_ref_map["dateTime"] = "https://schema.org/DateTime"
+    dataType_ref_map["date"] = "https://schema.org/Date"
+    return dataType_ref_map[type]
+
+
+def get_parameters(global_: Dict[str, Any],
+                   path: str, method: str, class_name: str)->str:
+    """
+    Parse paramters from method object
+    :param global_: global state
+    :param path: endpoint
+    :param method: method under consideration
     :param class_name: name of class
+    :return: param
+    """
+    param = str
+    for parameter in global_["doc"]["paths"][path][method]["parameters"]:
+        if allow_parameter(parameter):
+            try:
+                # check if class has been parsed
+                if parameter["schema"]["$ref"].split(
+                        '/')[2] in global_["class_names"]:
+                    param = "vocab:" + \
+                        parameter["schema"]["$ref"].split('/')[2]
+
+                else:
+                    # if not go to that location and parse and add
+                    get_class_details(
+                        global_,
+                        get_data_at_location(
+                            parameter["schema"]["$ref"]),
+                        parameter["schema"]["$ref"].split('/')[2],
+                        path=path)
+                    param = "vocab:" + \
+                            parameter["schema"]["$ref"].split('/')[2]
+            except KeyError:
+                type = parameter["type"]
+                if type == "array":
+                    # TODO change this after we find a way to represent array
+                    # in parameter using semantics
+                    items = parameter["schema"]["items"]
+                    try:
+                        if items["$ref"].split(
+                                '/')[2] in global_["class_names"]:
+                            param = "vocab" + items["$ref"].split('/')[2]
+                        else:
+                            get_class_details(
+                                global_, get_data_at_location(
+                                    items["$ref"]), items["$ref"].split('/')[2], path=path)
+                            param = "vocab" + items["$ref"].split('/')[2]
+                    except KeyError:
+                        param = type_ref_mapping(items["type"])
+                elif type == "object":
+                    print("parameter type object !!")
+                    print("converted to string")
+                    param = "string"
+                else:
+                    param = type_ref_mapping(type)
+
+        else:
+            print("cannot process the parameter")
+            print(parameter)
```

Comment:
Shouldn't this be inside a try-except block?

Reply:
im not expecting any exceptions , i just have to check if the param block follows the rules we have defined in allow_paramter